clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/HonoTopic.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.kafka.client;
+
+import java.util.Objects;
+
+import org.eclipse.hono.util.CommandConstants;
+import org.eclipse.hono.util.EventConstants;
+import org.eclipse.hono.util.TelemetryConstants;
+
+/**
+ * Identifier for Hono's topics. The Kafka topic string is obtained by {@link #toString()}.
+ */
+public final class HonoTopic {
+
+    private static final String SEPARATOR = ".";
+    private static final String NAMESPACE = "hono" + SEPARATOR;
+
+    private final Type type;
+    private final String tenantId;
+
+    /**
+     * Creates a new topic from the given topic type and tenant ID.
+     *
+     * @param type The type of the topic.
+     * @param tenantId The ID of the tenant that the topic belongs to.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public HonoTopic(final Type type, final String tenantId) {
+        Objects.requireNonNull(type);
+        Objects.requireNonNull(tenantId);
+
+        this.type = type;
+        this.tenantId = tenantId;
+    }
+
+    /**
+     * Creates a topic instance from the string representation.
+     *
+     * @param topicString The string to create a topic from.
+     * @return The topic or {@code null} if the string does not contain a valid Hono topic.
+     */
+    public static HonoTopic fromString(final String topicString) {
+        if (topicString.startsWith(Type.TELEMETRY.prefix)) {
+            return new HonoTopic(Type.TELEMETRY, topicString.substring(Type.TELEMETRY.prefix.length()));
+        } else if (topicString.startsWith(Type.EVENT.prefix)) {
+            return new HonoTopic(Type.EVENT, topicString.substring(Type.EVENT.prefix.length()));
+        } else if (topicString.startsWith(Type.COMMAND.prefix)) {
+            return new HonoTopic(Type.COMMAND, topicString.substring(Type.COMMAND.prefix.length()));
+        } else if (topicString.startsWith(Type.COMMAND_RESPONSE.prefix)) {
+            return new HonoTopic(Type.COMMAND_RESPONSE, topicString.substring(Type.COMMAND_RESPONSE.prefix.length()));
+        }
+        return null;
+    }
+
+    /**
+     * Gets the type of the topic.
+     *
+     * @return The type.
+     */
+    public Type getType() {
+        return type;
+    }
+
+    /**
+     * Gets the tenant ID of the topic.
+     *
+     * @return The tenant ID.
+     */
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    /**
+     * Returns the string representation of the topic as used by the Kafka client.
+     *
+     * @return The topic as a string.
+     */
+    @Override
+    public String toString() {
```

Comment:
this seems to be the most often used method, isn't it? Given that this class is immutable, we should probably create the string representation once during construction instead of concatenating the strings over and over again ...

Reply:
Done.